src/manual/multiple-peerconnections/index.html

```diff
@@ -0,0 +1,132 @@
+    this.remoteConnection = null;
+
+    var table = document.getElementById('test-table');
+    var row = table.insertRow(-1);
+
+    var localCell = row.insertCell(-1);
+    this.localView = document.createElement('video');
+    this.localView.autoplay = true;
+    localCell.appendChild(this.localView);
+
+    var remoteCell = row.insertCell(-1);
+    this.remoteView = document.createElement('video');
+    this.remoteView.autoplay = true;
+    remoteCell.appendChild(this.remoteView);
+
+    this.startTest = function() {
+      getUserMedia({audio: true, video: true},
+                   this.getUserMediaOkCallback.bind(this),
+                   this.getUserMediaFailureCallback.bind(this));
+    };
+
+    this.getUserMediaOkCallback = function(localStream) {
+      attachMediaStream(this.localView, localStream);
+      this.callUsingStream(localStream);
+    };
+
+    this.getUserMediaFailureCallback = function(error) {
+      console.error('getUserMedia request failed: name=' + error.name +
+                    ' constraintName=' + error.constraintName + ' message=' +
+                    error.message);
+    };
+
+    this.callUsingStream = function(localStream) {
+      this.localConnection = new RTCPeerConnection(null, null);
+      this.remoteConnection = new RTCPeerConnection(null, null);
+
+      this.localConnection.onicecandidate =
+        this.onIceCandidateToLocal.bind(this);
+      this.localConnection.addStream(localStream);
+
+      this.negotiate();
+    };
+
+    this.negotiate = function() {
+      this.localConnection.createOffer(
+          this.onOfferCreated.bind(this),
+          function(err) {});
+    };
+
+    this.onOfferCreated = function(offer) {
+      var self = this;
+      this.localConnection.setLocalDescription(offer, function() {
+        self.receiveCall(offer.sdp);
+      }, function(err) {});
+    };
+
+    this.receiveCall = function(offerSdp) {
+      var self = this;
+      this.remoteConnection.onicecandidate =
+        this.onIceCandidateToRemote.bind(this);
+      this.remoteConnection.onaddstream = function(e) {
+        attachMediaStream(self.remoteView, e.stream);
+      };
+
+      var parsedOffer = new RTCSessionDescription({ type: 'offer',
+                                                    sdp: offerSdp });
+      this.remoteConnection.setRemoteDescription(parsedOffer, function() {
+        self.remoteConnection.createAnswer(self.onAnswerCreated.bind(self),
+            ignore1);
+      }, ignore1);
+    };
+
+    this.onAnswerCreated = function(answer) {
+      var self = this;
+      this.remoteConnection.setLocalDescription(answer, function() {
+        self.handleAnswer(answer.sdp);
+      }, ignore1);
+    };
+
+    this.handleAnswer = function(answerSdp) {
+      var parsedAnswer = new RTCSessionDescription({ type: 'answer',
+                                                     sdp: answerSdp });
+      this.localConnection.setRemoteDescription(parsedAnswer, ignore1, ignore1);
```

Comment:
Use promise instead of callback, see https://github.com/webrtc/samples/blob/gh-pages/src/content/peerconnection/pc1/js/main.js#L140